lib/internal/fs/dir.js

```diff
@@ -293,13 +295,27 @@ class Dir {
   }
 }
 
-ObjectDefineProperty(Dir.prototype, SymbolAsyncIterator, {
-  __proto__: null,
-  value: Dir.prototype.entries,
+const nonEnumerableDescriptor = {
   enumerable: false,
   writable: true,
   configurable: true,
-});
+};
+ObjectDefineProperties(Dir.prototype, { __proto__: null,
+                                        [SymbolDispose]: {
+                                          __proto__: null,
+                                          ...nonEnumerableDescriptor,
+                                          value: Dir.prototype.closeSync,
+                                        },
+                                        [SymbolAsyncDispose]: {
+                                          __proto__: null,
+                                          ...nonEnumerableDescriptor,
+                                          value: Dir.prototype.close,
+                                        },
+                                        [SymbolAsyncIterator]: {
+                                          __proto__: null,
+                                          ...nonEnumerableDescriptor,
+                                          value: Dir.prototype.entries,
+                                        } });
```

Comment:
Nit: we don't need `__proto__: null` in the parent object because `Object.defineProperties` uses only own properties.
```suggestion
ObjectDefineProperties(Dir.prototype, {
  [SymbolDispose]: {
    __proto__: null,
    ...nonEnumerableDescriptor,
    value: Dir.prototype.closeSync,
  },
  [SymbolAsyncDispose]: {
    __proto__: null,
    ...nonEnumerableDescriptor,
    value: Dir.prototype.close,
  },
  [SymbolAsyncIterator]: {
    __proto__: null,
    ...nonEnumerableDescriptor,
    value: Dir.prototype.entries,
  },
});
```